apps/admin/backend/src/perf.test.ts

```diff
@@ -0,0 +1,70 @@
+import { computeFullElectionTally } from '@votingworks/utils';
+import { join } from 'path';
+import { electionMinimalExhaustiveSampleDefinition } from '@votingworks/fixtures';
+import { assert } from '@votingworks/basics';
+import { CastVoteRecord } from '@votingworks/types';
+import {
+  buildTestEnvironment,
+  configureMachine,
+  mockElectionManagerAuth,
+} from '../test/app';
+import { getBackupPath, takeBackup } from '../test/backups';
+import { getPerformanceTimer } from '../test/timer';
+
+// depending on the test conditions, may need to increase this
+jest.setTimeout(600000);
+
+const electionDefinition = electionMinimalExhaustiveSampleDefinition;
+
+test.skip('load CVR files', async () => {
+  const timer = getPerformanceTimer();
+  const { apiClient, workspace, auth } = buildTestEnvironment();
+  await configureMachine(apiClient, auth, electionDefinition);
+  mockElectionManagerAuth(auth, electionDefinition.electionHash);
+  timer.checkpoint(`setup complete`);
+
+  for (let i = 1; i <= 10; i += 1) {
+    // load reports generated with libs/cvr-fixture-generator to ../perf-fixtures
+    const reportDirectoryPath = join(
+      __dirname,
+      '../perf-fixtures',
+      '10000',
+      i.toString()
+    );
+    const addReportResult = await apiClient.addCastVoteRecordFile({
+      path: reportDirectoryPath,
+    });
+    assert(addReportResult.isOk());
+    timer.checkpoint(`file ${i} added`);
+  }
+
+  takeBackup({
+    workspace,
+    backupName: 'performance',
+  });
+});
+
+test.skip('tally performance', () => {
+  const timer = getPerformanceTimer();
+  const { workspace, auth } = buildTestEnvironment(
+    getBackupPath('performance')
```

Comment:
Would it work instead to do the common test env setup in a `beforeAll()` block? Or is this meant to facilitate persisting across multiple runs of the test file?

Reply:
The main motivation for the backup is that loading up the DB can take a long time. If I want 1,000,000 votes in the DB, that can take almost 20 minutes. I want to be able to do that once and then run the same test file multiple times, either to have a sample size larger than 1, to try variations in the code, or to repeatedly test destructive actions.

Reply:
Ah, this feature makes sense, then - thanks for explaining